### PR TITLE
docs: Update documentation for alpha commands

### PR DESCRIPTION
**Pull request by mortent**
Updates the reference docs on the kpt.dev site with the `kpt alpha repo` commands. This also uses the mdtogo tags to generate the cobra output for those commands from the site.

Wanted to get feedback on whether this is the right approach before doing the `rpkg` and `sync` commands.

We also need to figure out a good way to handle the flags about logging, talking to the apiserver, and printing. Explaining them all in every command seems unnecessary.

### CODE REVIEW

File: scripts/generate_site_sidebar/sidebar_template.md.tmpl
Comment (droot) [inline scripts/generate_site_sidebar/sidebar_template.md.tmpl:32]: I find `reg/unreg` unintuitive. should we use `repo add|remove` instead ?

File: site/reference/cli/alpha/repo/get/README.md
Comment (droot) [inline site/reference/cli/alpha/repo/get/README.md:13]: I am thinking if we should rename this to `list` :) 

`kpt alpha repo list` ?

**Comment by droot** [comment]
> We also need to figure out a good way to handle the flags about logging, talking to the apiserver, and printing. Explaining them all in every command seems unnecessary.

One way would adding a reference for `global flags` and linking to it from the commands that use global flags. And for command help, we might have to concatenate global flags help with the command help. WDYT ?

File: site/reference/cli/alpha/repo/get/README.md
Comment (martinmaly) [reply site/reference/cli/alpha/repo/get/README.md:13]: `kubectl` uses `get` though :(
also, `list` is an alias in place for `kpt alpha repo get` (https://github.com/GoogleContainerTools/kpt/blob/main/internal/cmdrepoget/command.go#L52)

**Comment by droot** [comment]
@mortent overall the approach looks good to me.

File: site/reference/cli/alpha/repo/get/README.md
Comment (droot) [reply site/reference/cli/alpha/repo/get/README.md:13]: Good to know that we have an alias :)

For some reason, my brain associates `get` with fetching single entity, but it could be just me. list alias solves my problem then so will stop bikeshedding :).

File: scripts/generate_site_sidebar/sidebar_template.md.tmpl
Comment (martinmaly) [reply scripts/generate_site_sidebar/sidebar_template.md.tmpl:32]: I'd recommend collecting all CLI surface questions and addressing them together. Renaming the reg/unreg or add/remove command is a good suggestion but since renames such as this are disruptive (we need to update docs, design doc, tests, ...) I'd recommend doing them at once.

**Review by martinmaly — APPROVED** [review]
lgtm though before merging we should figure out process to finalize the CLI surface.

**Comment by mortent** [comment]
@martinmaly Yeah, I plan to add docs for all the commands in this PR. Then we can figure out which changes to the API surface we want to make and then block this PR until we have addressed them.

**Comment by martinmaly** [comment]
Thank you, @mortent!

To your questions:
* until we introduce the Package resource in the future, I do believe that using language of `package revision` consistently is a good idea.
* +1 on the default mode for the `edit/copy` command which will create new revision in the same repository
* finalizing and publishing are the same. We used 'finalize' term first but switched to publish as a more commonly used term for that activity. So we should remove use of 'finalize' in favor of 'publish'.

**Comment by mortent** [comment]
@martinmaly Updated the PR to use package revision most places (still need some work here to clarify the terminology in some cases, but I'll do that in a follow-up), updated the docs for the copy command, and removed references to 'finalize' and instead use 'publish'.

File: commands/repocmd.go
Comment (martinmaly) [inline commands/repocmd.go:37]: this should probably be `"[Alpha] " + repodocs.RepoXYZ` so we get a space between [Alpha] and the doc comment

File: site/reference/cli/alpha/repo/unreg/README.md
Comment (martinmaly) [inline site/reference/cli/alpha/repo/unreg/README.md:34]: super nit: I'd break line before `By`

File: site/reference/cli/alpha/rpkg/copy/README.md
Comment (martinmaly) [inline site/reference/cli/alpha/rpkg/copy/README.md:15]: `name or revision` ?

(since we are considering the default behavior to keep name the same and only rev the revision)

**Review by martinmaly — APPROVED** [review]
Only a couple of nits, they can be handled in follow-up.

File: scripts/generate_site_sidebar/sidebar_template.md.tmpl
Comment (droot) [reply scripts/generate_site_sidebar/sidebar_template.md.tmpl:32]: Ack.

**Review by droot — APPROVED** [review]

File: commands/repocmd.go
Comment (mortent) [reply commands/repocmd.go]: Good catch. Fixed.

File: site/reference/cli/alpha/rpkg/copy/README.md
Comment (mortent) [reply site/reference/cli/alpha/rpkg/copy/README.md]: Agree revision is a better word here. Updated.

**Pull request merged**
